server/admin/http/handler.go

```diff
@@ -394,3 +404,50 @@ func decodePeerID(id string, w http.ResponseWriter) (peer.ID, bool) {
 	}
 	return peerID, true
 }
+
+func healthCheckValueStore(h *adminHandler) (cause ValueStoreFailReason, err error) {
+	mhash, val, err := createValueStoreDatum()
+	if err != nil {
+		return VS_GEN_FAILURE, err
+	}
+
+	err = h.indexer.Put(*val, mhash)
+	if err != nil {
+		return VS_WRITE_FAILURE, err
+	}
+
+	defer func() {
+		if errorOnRemove := h.indexer.Remove(*val, mhash); errorOnRemove != nil {
+			cause = VS_REMOVE_FAILURE
+			err = errorOnRemove
+		}
+	}()
+
+	rval, present, err := h.indexer.Get(mhash)
+
+	if err != nil {
+		return VS_READ_FAILURE, err
+	}
+	if !present {
+		return VS_NO_DATA, err
+	}
+	if !val.Equal(rval[0]) {
+		return VS_READWRITE_MISMATCH, err
+	}
+
+	return "", nil
```

Comment:
I was trying to introduce the concept of an error code but since these messages won't be used anywhere else, makes sense to go with just messages. Made necessary changes.